DataManagementSystem/DB/FileCatalogComponents/DirectoryLevelTree.py

```diff
@@ -1,124 +1,136 @@
-
-""" DIRAC FileCatalog component representing a directory tree with 
-    enumerated paths
+""" DIRAC FileCatalog component representing a directory tree with enumerated paths
 """
 
+# pylint: disable=protected-access
+
 
 __RCSID__ = "$Id$"
 
 import os
-from types import ListType, StringTypes
 from DIRAC import S_OK, S_ERROR
 from DIRAC.DataManagementSystem.DB.FileCatalogComponents.DirectoryTreeBase import DirectoryTreeBase
 
 MAX_LEVELS = 15
 
+
 class DirectoryLevelTree(DirectoryTreeBase):
-  """ Class managing Directory Tree as a simple self-linked structure 
-      with full directory path stored in each node 
+  """ Class managing Directory Tree as a simple self-linked structure
+      with full directory path stored in each node
   """
-  
-  def __init__(self,database=None):
-    DirectoryTreeBase.__init__(self,database)
+
+  def __init__(self, database=None):
+    DirectoryTreeBase.__init__(self, database)
     self.treeTable = 'FC_DirectoryLevelTree'
 
   def getTreeType(self):
-    
+
     return 'Directory'
 
-  def findDir(self,path,connection=False):
+  def findDir(self, path, connection=False):
     """  Find directory ID for the given path
     """
-    
-    dpath = os.path.normpath( path )    
+
+    dpath = self.db._escapeString(os.path.normpath(path))
+    if not dpath['OK']:
+      return dpath
+    dpath = dpath['Value']
     req = "SELECT DirID,Level from FC_DirectoryLevelTree WHERE DirName='%s'" % dpath
-    result = self.db._query(req,connection)
+    result = self.db._query(req, connection)
     if not result['OK']:
       return result
-    
+
     if not result['Value']:
       return S_OK('')
-    
-    res = S_OK( result['Value'][0][0] )
+
+    res = S_OK(result['Value'][0][0])
     res['Level'] = result['Value'][0][1]
     return res
-  
-  def findDirs( self, paths, connection=False ):
+
+  def findDirs(self, paths, connection=False):
     """ Find DirIDs for the given path list
     """
-    dpaths = ','.join( [ "'"+os.path.normpath( path )+"'" for path in paths ] )
+    dpaths = ','.join(["'" + os.path.normpath(path) + "'" for path in paths])
+    dpaths = self.db._escapeString(dpaths)
+    if not dpaths['OK']:
+      return dpaths
+    dpaths = dpaths['Value']
     req = "SELECT DirName,DirID from FC_DirectoryLevelTree WHERE DirName in (%s)" % dpaths
-    result = self.db._query(req,connection)
+    result = self.db._query(req, connection)
     if not result['OK']:
       return result
     dirDict = {}
     for dirName, dirID in result['Value']:
       dirDict[dirName] = dirID
 
-    return S_OK( dirDict )
-  
-  def removeDir(self,path):
+    return S_OK(dirDict)
+
+  def removeDir(self, path):
     """ Remove directory
     """
 
     result = self.findDir(path)
     if not result['OK']:
-      return result   
+      return result
     if not result['Value']:
       res = S_OK()
       res["DirID"] = 0
       return res
-    
+
     dirID = result['Value']
+    dirID = self.db._escapeString(dirID)
```

Comment:
I am guessing dirID is an integer before it is passed to escapeString, after which it is a string, which will case an exception in line 84
` ... %d" % dirID `